src/botmation/helpers/pipe.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Functions that help with the Pipe
+ */
+
+import { isPiped, Piped } from "botmation/types/piped"
+
+/**
+ * @description    Returns an array of all the injects with the piped value unpiped (removed from branded wrapping)
+ *    Can't remap `value` with restructuring when it's undefined
+ * @param injectsPiped 
+ * @todo move to a pipe helper
+ */
+export const openInjectsPipe = (injectsPiped: any[]): any[] => {
+  let injectsWithoutPipe = injectsPiped.slice(0, injectsPiped.length - 1)
+
+  // if the branded pipe has a value, return that, otherwise return undefined
+  return [...injectsWithoutPipe, getInjectsPipedValue(injectsPiped)]

```

Comment:
If no piped value, then missing 1 inject from above slice and this will error

Add safety check

Reply:
`unpipeInjects()` safe way to extract injects that may or may not have appended piped value